sympy/vector/operators.py

```diff
@@ -0,0 +1,154 @@
+from sympy.vector.coordsysrect import CoordSysCartesian
+from sympy.core import Basic
+from sympy.vector.vector import Vector
+from sympy.vector.deloperator import Del
+
+
+class Gradient(Basic):
+    """
+    Represents unevaluated Gradient.
+    
+    Examples
+    ========
+    
+    >>> from sympy.vector import CoordSysCartesian, Gradient
+    >>> R = CoordSysCartesian('R')
+    >>> s1 = R.x*R.y*R.z
+    >>> Gradient(s1, R)
+    (Derivative(R.x*R.y*R.z, R.x))*R.i git + (Derivative(R.x*R.y*R.z, R.y))*R.j + (Derivative(R.x*R.y*R.z, R.z))*R.k
+    """
+
+    def __new__(cls, scalar, coord_sys):
+        return Del(coord_sys).gradient(scalar)
+
+
+class Divergence(Basic):
+    """
+    Represents unevaluated Divergence.
+
+    Examples
+    ========
+    
+    >>> from sympy.vector import CoordSysCartesian, Gradient
+    >>> R = CoordSysCartesian('R')
+    >>> v = R.x*R.y*R.z*R.i + R.j + R.k
+    >>> Divergence(v, R)
+    Derivative(R.x*R.y*R.z, R.x)
+    
+    """
+
+    def __new__(cls, vect, coord_sys):
+        return Del(coord_sys).dot(vect)
+
+
+class Curl(Basic):
+    """
+    Represents unevaluated Curl.
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, Curl
+    >>> R = CoordSysCartesian('R')
+    >>> s1 = R.x*R.y*R.z*R.i + R.j
+    Curl(s1, R)
+    (Derivative(0, R.y) - Derivative(1, R.z))*R.i + (-Derivative(0, R.x) + 
+    Derivative(R.x*R.y*R.z, R.z))*R.j + (Derivative(1, R.x) - Derivative(R.x*R.y*R.z, R.y))*R.k    
+    
+    """
+
+    def __new__(cls, vect, coord_sys):
+        return Del(coord_sys).cross(vect)
+
+
+def gradient(scalar, coord_sys):
```

Comment:
Default argument `coord_sys=None`, then put an `if coord_sys == None:` in the function code. Same for the other functions.